### DISCUSSION TITLE
Existing semantic of REGO query language

### DISCUSSION THREAD

**Discussion by massimilianobaldo** [open-policy-agent/community#634] [OPA and Rego]
Hi everybody, I have a question about the REGO language.
In particular, I want to know if there exists any kind of formalization or semantic about this query language.
I'm working in a founded project for an Univeristy in Italy and we want to investigate more in the possible usages of REGO.
Also, if you can guide me in which sections of the codebase (i.e folders) to analyze for the checking phase of REGO rules. 
Thanks evreybody!

**Comment by anderseknert** [comment]
Hi there @massimilianobaldo 👋 And thanks for showing interest in OPA! The [policy language](https://www.openpolicyagent.org/docs/latest/policy-language/) and [policy reference](https://www.openpolicyagent.org/docs/latest/policy-reference/) sections of the docs are probably good places to start. The grammar is described [here](https://www.openpolicyagent.org/docs/latest/policy-reference/#grammar), although it could be it's not 100% complete.

If you're interested in the details of evaluation specifically, perhaps @johanfylling can provide some pointers.
